CorsixTH/Lua/languages/english.lua

```diff
@@ -269,6 +269,10 @@ dynamic_info.staff.actions.fired = "Fired"
 dynamic_info.staff.actions.vaccine = "Vaccinating a patient"
 dynamic_info.patient.actions.epidemic_vaccinated = "I am no longer contagious"
 
+if self.ui.app.config.new_machine_extra_info then
+  dynamic_info.object.strength = "Strength %d (Replacement %d)"
+end
+
```

Comment:
Logic does not belong inside a language file, it'll need to be in `machine.lua`; this may mean you'll need two versions of the string, or break it down (since I believe `dynamic_info.object.strength` exists) so your new part of the string is 
`dynamic_info.object.strength_extra_info = " (Replacement %d)"` -- note the space.
On the text part, maybe you could have "(Upgradeable to %d)" instead? Or "(Replaceable to %d)"?

Then you can join them together
```lua
local strength_text = (_S.dynamic_info.object.strength):format(self.strength)
if hosp.ui.app.config.new_machine_extra_info then
  local start_strength = hosp.research.research_progress[self.object_type].start_strength
  strength_text = strength_text .. (_S.dynamic_info.object.strength_extra_info):format(start_strength)
end
self:setDynamicInfo("text", {
  self.object_type.name,
  strength_text,
  _S.dynamic_info.object.times_used:format(self.times_used),
})
```
Not that you have to use that... but if you do I may have made a booboo somewhere writing on the fly.

Reply:
Some languages might take issue with how strings are concatenated in the code. Perhaps in the future the strings in the language lua files can dictate which variable value is placed where inside the text to be formatted.